sync/src/chain.rs

```diff
@@ -2018,19 +2017,13 @@ impl ChainSync {
 			self.restart(io);
 		}
 
-		if !is_syncing && !enacted.is_empty() {
-			// Select random peers to re-broadcast transactions to.
-			let mut random = random::new();
-			let len = self.peers.len();
-			let peers = random.gen_range(0, min(len, 3));
-			trace!(target: "sync", "Re-broadcasting transactions to {} random peers.", peers);
-
-			for _ in 0..peers {
-				let peer = random.gen_range(0, len);
-				self.peers.values_mut().nth(peer).map(|mut peer_info| {
-					peer_info.last_sent_transactions.clear()
-				});
-			}
+		if !is_syncing && !enacted.is_empty() && self.peers.is_empty() {
+			// Select random peer to re-broadcast transactions to.
+			let peer = random::new().get_rng(0, self.peers.len());
+			trace!(target: "sync", "Re-broadcasting transactions to random peer.");
+			self.peers.get_mut(peer).map(|mut peer_info|
```

Comment:
I think `PeerId`s are not bound to `self.peers.len()`, it could be arbitrary `usize`, so most of the time this will not clear any peer.